### PR TITLE
Upgrade code & environment for JDK 17.  But not using it yet.

### PR DESCRIPTION
**Pull request by fhcarter**
Upgrade source to JDK 17 level without warnings.

In prep for some other work to be done soon, upgrade things to run with JDK 17.  This is mostly changes to use of the Java security manager (which is now deprecated) that we were using in tests.  Replaced with try-finally & a so-called `ExitProcessor` that allows us to through exceptions rather than exit (useful in testing).  This is a test-only change.

Upgraded/changed Lombok plugin since the one in use is in maintenance-only mode and doesn't support Java 17.

Other minor tweaks.

### CODE REVIEW

File: CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVMain.java [hunk 1]
```diff
@@ -41,6 +41,8 @@ public class CSVMain {
     static final int NO_AUTH_EXIT = 1;
     static final int NO_SOURCE_EXIT = 2;
     static final int NO_SERVER_EXIT = 3;
+
```
Comment (fhcarter) [inline CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVMain.java:44]: this is a common pattern.  It's seen in many of the connectors that used the security manager

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -1,7 +1,6 @@
-
 plugins {
     id 'java'
```
Comment (fhcarter) [inline buildSrc/build.gradle:2]: old Lombok processor no longer maintained, and doesn't work with JDK 17.  Upgrade to IntelliJ recommended one.  Works fine w/o code changes (other than build.gradle code)

File: buildSrc/build.gradle [hunk 3]
```diff
@@ -12,7 +11,7 @@ ext {
     apacheCommonsLangVersion = '3.12.0'
     jacksonVersion = '2.14.2'
     log4JVersion='2.17.2'
```
Comment (fhcarter) [inline buildSrc/build.gradle:13]: use common version throughout.

File: jdbcSource/build.gradle [hunk 4]
```diff
@@ -1,10 +1,14 @@
-group 'io.vantiq'
-version 'unspecified'
```
Comment (fhcarter) [inline jdbcSource/build.gradle:1]: Using new-style plugins closure requires moving it in gradle file.  No substantive changes here other than the new Lombok plugin

**Comment by fhcarter** [comment]
[Jenkins run ](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/256/)

**Comment by fhcarter** [comment]
(Changes tested using Javas 11 & 17.  17 required for a separate upgrade (to be commenced soon))

**Review by mer2github — APPROVED** [review]
LGTM

**Pull request merged**